pageserver/src/tenant/storage_layer/layer.rs

```diff
@@ -426,6 +430,13 @@ struct LayerInner {
     /// For loaded layers (resident or evicted) this comes from [`LayerFileMetadata::generation`],
     /// for created layers from [`Timeline::generation`].
     generation: Generation,
+
+    /// The shard of this Layer
+    ///
+    /// For created layers, this is always the shard of the parent creating [`crate::tenant::Tenant`] object.  For
+    /// loaded layers, this may be some other shard, if the tenant has undergone a shard split
+    /// since the layer was originally written.
+    shard: ShardIndex,
```

Comment:
```suggestion
    /// The shard of this Layer.
    ///
    /// For layers created in this process, this will always be the [`ShardIndex`] of the
    /// current `ShardIdentity`` (TODO: add link once it's introduced).
    ///
    /// For loaded layers, this may be some other value if the tenant has undergone
    /// a shard split since the layer was originally written.
    shard: ShardIndex,
```